### ISSUE TITLE
Allman brace style is not honored for switch statements

### ISSUE THREAD

**Issue by wabain** [eslint/eslint#3903]
I would expect that if `"allman"` is specified for `brace-style`, ESLint would ensure that the opening brace of a switch statement is on a new line. However, ESLint [always](https://github.com/eslint/eslint/blob/0b3732bf3bbefacb3ccf0a4436a9b4b6139bd4b2/lib/rules/brace-style.js#L181) ensures that the brace is on the same line as the controlling statement. In ESLint 1.5.1, given the input:

```js
/* eslint brace-style: [2, "allman"] */

switch (1)
{
    case 1:
       break;
}

```

ESLint outputs the error:

    3:1  error  Opening curly brace does not appear on the same line as controlling statement  brace-style

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Verified this in the demo. It looks like switch statement is not covered for some reason.

**Comment by gyandeeps** [comment]
Working on this

**Comment by aparajita** [comment]
@ilyavolodin Comparing with "1tbs" is wrong, because "stroustrop" then fails to do the right thing. #4003 is correct.
